source/braille.py

```diff
@@ -1147,7 +1148,7 @@ def _setCursor(self, info: textInfos.TextInfo):
 
 	def _getTypeformFromFormatField(self, field, formatConfig):
 		typeform = louis.plain_text
-		if not formatConfig["reportFontAttributes"]:
+		if not (formatConfig["fontAttributeReporting"] & OutputMode.BRAILLE):
```

Comment:
When dealing with config in the past, I was told not to use bitwise testing / masking for such high level Python code. This is especially relevant here since there is nothing explicit in `OutputMode` enum definition indicating that it should honour such bitwise behaviour.

The following would be more in line with NVDA's existing code style and Python code style:

```suggestion
		if formatConfig["fontAttributeReporting"] not in (OutputMode.BRAILLE, OutputMode.SPEECH_AND_BRAILLE):
```

Reply:
Personally, I don't like this solution. I feel that it isn't DRY. However, I agree that the code is currently a little unclear. Would we be happy if I switched to using `DisplayStringFlag` for this instead? Then The if statements could be something like
```python
if OutputMode.BRAILLE not in formatConfig["fontAttributeReporting"]:
```
@CyrilleB79, @seanbudd, @michaelDCurran?

Reply:
>This is especially relevant here since there is nothing explicit in OutputMode enum definition indicating that it should honour such bitwise behaviour.

@CyrilleB79 - I've made a suggestion here to make this more explicit: https://github.com/nvaccess/nvda/pull/16748#discussion_r1665176647

Reply:
@SaschaCowley sorry, I had missed your reply.

What do you mean with "it isn't dry"?

Re your proposal, do you suggest to check if the string "braille" is in the config value? It seems worse to me.
An other option is to define `isBrailleOutput` and `isSpeechOutput` methods in the enum class and to use these explicit function:
```suggestion
		if not OutputMode(formatConfig["fontAttributeReporting"]).isBrailleOutput():
```

I did not test this, so that may be wrong, but it's just to give the idea.

Reply:
> What do you mean with "it isn't dry"?

DRY stands for "don't repeat yourself". It feels to me like saying `x in (this, this_and_that)` is repeating myself, as all I am interested in is if the `this` bit is true.

> Re your proposal, do you suggest to check if the string "braille" is in the config value? It seems worse to me.

`DisplayStringFlag` is based on `enum.Flag`, which is a bit mask. The `in` syntax for such flags is essentially equivalent to 

```python
def__contains__(self, rhs):
    return self & rhs != 0
```

Which is essentially just syntactic sugar for what I was doing.

> An other option is to define isBrailleOutput and isSpeechOutput methods in the enum class and to use these explicit function:

This is what I ended up doing, as using `Flag` or `IntFlag` subclasses introduces issues with iterating over enum values. If we end up switching to a GUI which doesn't require iterating over the possible values (separate checkboxes for speech and braille, for example), we can consider switching to an `IntFlag`, which theoretically shouldn't break anything except the iterability of all values including aliases.